src/azure/storage/signer.rs

```diff
@@ -69,7 +69,16 @@ impl Signer {
             }
             Credential::SharedKey(ak, sk) => match method {
                 SigningMethod::Query(_) => {
```

Comment:
We can use user's expire.

Reply:
could you please tell me how to set user's expire? get it from ctx?

Reply:
The value in `SigningMethod::Query(v)` is a duration which is the users expires_in.

Reply:
Wow, that's an amazing design! thank you lmk, I'll do it now.

Reply:
I noticed that we set the expiration time to 1 second here. Do we need to increase this time?
https://github.com/godruoyi/reqsign/blob/55d179e3e4c5fcbbd70a6e3bf6108506e3e29852/src/azure/storage/signer.rs#L140

Reply:
We need to change this API into `sign_query(req: &mut impl SignableRequest, cred: &Credential, expire: Duration)` like `s3` and `gcs`.